framework/source/class/qx/Promise.js

```diff
@@ -0,0 +1,6551 @@
+     * use it over Promise.all (which awaits for the entire array before you can call Array#reduce on it).
+     * 
+     * @param iterable {Iterable} An iterable object, such as an Array
+     * @param reducer {Function} the callback, with <code>(value, index, length)</code>
+     * @param initialValue {Object?} optional initial value
+     * @return {qx.Promise}
+     */
+    reduce: function(iterable, reducer, initialValue) {
+      return this.__callStaticMethod('reduce', arguments);
+    },
+
+    /**
+     * Returns a new function that wraps the given function fn. The new function will always return a promise that is 
+     * fulfilled with the original functions return values or rejected with thrown exceptions from the original function.
+     * @param cb {Function}
+     * @return {Function}
+     */
+    method: function(cb) {
+      var wrappedCb = qx.Promise.Bluebird.method(cb);
+      return function() {
+        return qx.Promise.__wrap(wrappedCb.apply(this, arguments));
+      };
+    },
+    
+    /**
+     * Like .all but for object properties or Maps* entries instead of iterated values. Returns a promise that 
+     * is fulfilled when all the properties of the object or the Map's' values** are fulfilled. The promise's 
+     * fulfillment value is an object or a Map with fulfillment values at respective keys to the original object 
+     * or a Map. If any promise in the object or Map rejects, the returned promise is rejected with the rejection 
+     * reason.
+     * 
+     * If object is a trusted Promise, then it will be treated as a promise for object rather than for its 
+     * properties. All other objects (except Maps) are treated for their properties as is returned by 
+     * Object.keys - the object's own enumerable properties.
+     * 
+     * @param input {Object} An Object
+     * @return {qx.Promise}
+     */
+    props: function(input) {
+      return this.__callStaticMethod('props', arguments);
+    },
+    
+    
+    
+    /* *********************************************************************************
+     * 
+     * Internal API methods
+     * 
+     */
+    
+    /**
+     * Called when the Bluebird Promise class is loaded
+     * @param Promise {Class} the Promise class
+     */
+    __attachBluebird: function(Promise) {
+      this.Bluebird = Promise;
+      Promise.config({
+        warnings: qx.core.Environment.get("qx.promise.warnings"),
+        longStackTraces: qx.core.Environment.get("qx.promise.longStackTraces"),
+        cancellation: true
+      });
+    },
+
+    /** Whether one-time initialisaton has happened */
+    __initialized: false,
+    
+    /**
+     * One-time initializer
+     */
+    __initialize: function() {
+      if (this.__initialized) {
+        return;
+      }
+      this.__initialized = true;
+      window.addEventListener("unhandledrejection", this.__onUnhandledRejection.bind(this));
```

Comment:
@level420 sure, but I think that those kind of checks should be isolated into a separate environment check, so that there's one, definitive way of checking.  I tend to agree with @derrell that we should look at whether `qx.bom.Event.addNativeListener` should be deciding whether to add a `process.on()` when node is detected, but that IMHO that should be part of a separate PR.  If we do agree to go down that route, then the latest commit would be fixed too 😀